src/auth/auth.service.ts

```diff
@@ -88,7 +88,7 @@ export class AuthService {
         `User '${user.userName}' has already 200 tokens and can't have anymore`,
       );
     }
-    const secret = await this.randomString(64);
+    const secret = this.BufferToBase64Url(await this.randomString(64));
     const keyId = this.BufferToBase64Url(await this.randomString(8));
     const accessTokenString = await this.hashPassword(secret.toString());
     const accessToken = this.BufferToBase64Url(Buffer.from(accessTokenString));
```

Comment:
By now the validate never matches. This can be fixed by removing this line, but the output in database is not base64 encoded anymore. An alternative is to decode the string before the comparison with the token which was send be the client. 
```suggestion
```